src/zmalloc.c

```diff
@@ -87,10 +87,22 @@ void zlibc_free(void *ptr) {
 #define dallocx(ptr,flags) je_dallocx(ptr,flags)
 #endif
 
-#define update_zmalloc_stat_alloc(__n) atomicIncr(used_memory,(__n))
-#define update_zmalloc_stat_free(__n) atomicDecr(used_memory,(__n))
+#define update_zmalloc_stat_alloc(__n) used_memory_tls[thread_index] += (__n)
+#define update_zmalloc_stat_free(__n) used_memory_tls[thread_index] -= (__n)
```

Comment:
```suggestion
#define update_zmalloc_stat_alloc(__n) used_memory_thread[thread_index] += (__n)
#define update_zmalloc_stat_free(__n) used_memory_thread[thread_index] -= (__n)
```
One of my greatest pet peeves is naming thread local storage TLS, because my mind only thinks it means transport layer security :)

Reply:
yeah TLS is unfortunatelly too popular :). `_thread` works for me too.